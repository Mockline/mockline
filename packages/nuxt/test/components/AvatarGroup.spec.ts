import { describe, it, expect } from 'vitest'
import ComponentRender from '../component-render'
import type { TypeOf } from 'zod'
// @ts-ignore
import { MAvatarGroup, MAvatar } from '#components'

describe('AvatarGroup', () => {
  it.each([
    ['basic group', {
      slots: {
        default: () => `
          <MAvatar src="https://avatars.githubusercontent.com/u/72015679?v=4" alt="Avatar" />
          <MAvatar src="https://avatars.githubusercontent.com/u/5158436?v=4" alt="Avatar" />
          <MAvatar src="https://avatars.githubusercontent.com/u/71938701?v=4" alt="Avatar" />
        `
      }
    }],
    ['with max limit', {
      props: {
        max: 2,
        size: "md"
      },
      slots: {
        default: () => `
          <MAvatar src="https://avatars.githubusercontent.com/u/72015679?v=4" alt="Avatar" />
          <MAvatar src="https://avatars.githubusercontent.com/u/5158436?v=4" alt="Avatar" />
          <MAvatar src="https://avatars.githubusercontent.com/u/71938701?v=4" alt="Avatar" />
          <MAvatar />
          <MAvatar src="https://avatars.githubusercontent.com/u/71938701?v=4" alt="Avatar" />
        `
      }
    }],
    ['with exact max', {
      props: {
        max: 4,
        size: "md"
      },
      slots: {
        default: () => `
          <MAvatar src="https://avatars.githubusercontent.com/u/72015679?v=4" alt="Avatar" />
          <MAvatar src="https://avatars.githubusercontent.com/u/5158436?v=4" alt="Avatar" />
          <MAvatar src="https://avatars.githubusercontent.com/u/71938701?v=4" alt="Avatar" />
          <MAvatar src="https://avatars.githubusercontent.com/u/71981971?v=4" alt="Avatar" />
        `
      }
    }],
    ['with fewer than max', {
      props: {
        max: 4,
        size: "md"
      },
      slots: {
        default: () => `
          <MAvatar src="https://avatars.githubusercontent.com/u/72015679?v=4" alt="Avatar" />
          <MAvatar src="https://avatars.githubusercontent.com/u/71938701?v=4" alt="Avatar" />
        `
      }
    }],
    ['with larger size', {
      props: {
        max: 4,
        size: "xl"
      },
      slots: {
        default: () => `
          <MAvatar src="https://avatars.githubusercontent.com/u/72015679?v=4" alt="Avatar" />
          <MAvatar src="https://avatars.githubusercontent.com/u/71938701?v=4" alt="Avatar" />
        `
      }
    }],
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof MAvatarGroup.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MAvatarGroup)
    expect(html).toMatchSnapshot()
  })
})
