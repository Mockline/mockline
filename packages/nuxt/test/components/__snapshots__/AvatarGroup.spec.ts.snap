// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AvatarGroup > renders basic group correctly 1`] = `
"<div class="flex items-center -space-x-[var(--overlap)] relative" style="--overlap: 0.25rem;">
  <!-- Display the visible avatars -->
  <!-- Display the remainder count -->
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with exact max correctly 1`] = `
"<div class="flex items-center -space-x-[var(--overlap)] relative" style="--overlap: 0.25rem;">
  <!-- Display the visible avatars -->
  <!-- Display the remainder count -->
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with fewer than max correctly 1`] = `
"<div class="flex items-center -space-x-[var(--overlap)] relative" style="--overlap: 0.25rem;">
  <!-- Display the visible avatars -->
  <!-- Display the remainder count -->
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with larger size correctly 1`] = `
"<div class="flex items-center -space-x-[var(--overlap)] relative" style="--overlap: 0.25rem;">
  <!-- Display the visible avatars -->
  <!-- Display the remainder count -->
  <!--v-if-->
</div>"
`;

exports[`AvatarGroup > renders with max limit correctly 1`] = `
"<div class="flex items-center -space-x-[var(--overlap)] relative" style="--overlap: 0.25rem;">
  <!-- Display the visible avatars -->
  <!-- Display the remainder count -->
  <!--v-if-->
</div>"
`;
